pallets/living-assets-evolution/src/lib.rs

```diff
@@ -103,15 +131,99 @@ pub mod pallet {
+			let counter = collection_id.checked_add(One::one()).ok_or(ArithmeticError::Overflow)?;
 			CollectionCounter::<T>::put(counter);
 
 			// Emit an event.
 			Self::deposit_event(Event::CollectionCreated { collection_id, owner: who });
 
-			// Return a successful DispatchResultWithPostInfo
+			// Return a successful DispatchResult
 			Ok(())
 		}
+
+		/// Mint new asset with external URI
+		///
+		/// This function performs the minting of a new asset with setting its external URI.
+		///
+		/// NOTE: This function will panic if the `slot` has a value greater than `2^96 - 1`
+		/// This will be fixed in the future https://github.com/freeverseio/laos-evolution-node/issues/77
+		///
+		/// # Errors
+		///
+		///  This function returns a dispatch error in the following cases:
+		///
+		/// * [`NoPermission`](`Error::<T>::NoPermission`) - if the caller is not the owner of the collection
+		/// * [`CollectionDoesNotExist`](`Error::<T>::CollectionDoesNotExist`) - if the collection does not exist
+		/// * [`AlreadyMinted`](`Error::<T>::AlreadyMinted`) - if the asset is already minted
+		/// * [`Overflow`](`ArithmeticError::<T>::Overflow`) - if the `slot` is greater than `2^96 - 1`
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::mint_with_external_uri())]
+		pub fn mint_with_external_uri(
+			origin: OriginFor<T>,
+			collection_id: CollectionId,
+			slot: Slot,
+			to: AccountIdOf<T>,
+			token_uri: TokenUriOf<T>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			ensure!(
+				CollectionOwner::<T>::contains_key(collection_id),
+				Error::<T>::CollectionDoesNotExist
+			);
+
+			ensure!(
+				CollectionOwner::<T>::get(collection_id) == Some(who),
+				Error::<T>::NoPermission
+			);
+
+			// compose asset_id	from slot and owner
+			let token_id = Self::slot_and_owner_to_token_id((slot, to.clone()));
+
+			ensure!(
+				TokenURI::<T>::get(collection_id, token_id).is_none(),
+				Error::<T>::AlreadyMinted
+			);
+
+			// Slot must be 96 bits
+			// TODO: use a custom type for this https://github.com/freeverseio/laos-evolution-node/issues/77
+			ensure!(slot <= MAX_U96, ArithmeticError::Overflow);
+
+			TokenURI::<T>::insert(collection_id, token_id, token_uri.clone());
+
+			Self::deposit_event(Event::MintedWithExternalTokenURI {
+				collection_id,
+				slot,
+				to,
+				token_id,
+				token_uri,
+			});
+
+			Ok(())
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	// Utility functions
+	/// A struct responsible for converting `Slot` and `AccountId` to `TokenId`
+	///
+	/// Every slot is identified by a unique `token_id` where `token_id = concat(slot #, owner_address)`
+	fn slot_and_owner_to_token_id(slot_and_owner: (Slot, AccountIdOf<T>)) -> TokenId {
+		let (slot, owner) = slot_and_owner;
+
+		let mut bytes = [0u8; 32];
+
+		let slot_bytes = slot.to_be_bytes();
+
+		// NOTE: this will panic at runtime if two arrays overlap, we should see if there is a safer way to do this
```

Comment:
Could you please clarify how the two arrays might overlap, given that they are declared as non-overlapping just above? There seems to be a contradiction.